src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/XamlNamespaceUpgradeStep.cs

```diff
@@ -0,0 +1,173 @@
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Maui
+{
+    public class XamlNamespaceUpgradeStep : UpgradeStep
+    {
+        private static readonly IReadOnlyDictionary<string, string> XamarinToMauiReplacementMap = new Dictionary<string, string>
+        {
+            {"http://xamarin.com/schemas/2014/forms", "http://schemas.microsoft.com/dotnet/2021/maui"},
+            {"http://xamarin.com/schemas/2020/toolkit", "http://schemas.microsoft.com/dotnet/2022/maui/toolkit"},
+            {"clr-namespace:Xamarin.Forms.PlatformConfiguration.AndroidSpecific;assembly=Xamarin.Forms.Core", "clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.AndroidSpecific;assembly=Microsoft.Maui.Controls"},
+            {"clr-namespace:Xamarin.Forms.PlatformConfiguration.iOSSpecific;assembly=Xamarin.Forms.Core", "clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.iOSSpecific;assembly=Microsoft.Maui.Controls"},
+            {"clr-namespace:Xamarin.Forms.PlatformConfiguration.macOSSpecific;assembly=Xamarin.Forms.Core", "clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.macOSSpecific;assembly=Microsoft.Maui.Controls"},
+            {"clr-namespace:Xamarin.Forms.PlatformConfiguration.TizenSpecific;assembly=Xamarin.Forms.Core", "clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.TizenSpecific;assembly=Microsoft.Maui.Controls"},
+            {"clr-namespace:Xamarin.Forms.PlatformConfiguration.WindowsSpecific;assembly=Xamarin.Forms.Core", "clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.WindowsSpecific;assembly=Microsoft.Maui.Controls"},
+        };
+
+        private readonly IPackageRestorer _restorer;
+
+        public override string Title => "Update XAML Namespaces";
+
+        public override string Description => "Updates XAML namespaces to .NET MAUI";
+
+        public XamlNamespaceUpgradeStep(IPackageRestorer restorer, ILogger<XamlNamespaceUpgradeStep> logger)
+            : base(logger)
+        {
+            _restorer = restorer;
+        }
+
+        public override IEnumerable<string> DependsOn { get; } = new[]
+        {
+            WellKnownStepIds.TemplateInserterStepId,
+        };
+
+        public override IEnumerable<string> DependencyOf { get; } = new[]
+        {
+            WellKnownStepIds.NextProjectStepId,
+        };
+
+        protected override async Task<UpgradeStepApplyResult> ApplyImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var project = context.CurrentProject.Required();
+            var roslynProject = GetBestRoslynProject(project.GetRoslynProject());
+            var solution = roslynProject.Solution;
+            int fileCount = 0;
+            int errorCount = 0;
+
+            // MSBuildWorkspace does not allow AdditionalDocument changes before MicrosoftCodeAnalysisVersion 4.0.0,
```

Comment:
@mjrousos Do you remember we you had to pin the Roslyn version down to 3.8.0 with the following comment?
> Use version 3.8.0 of CodeAnalysis assemblies to preserve compatibility with Visual Studio 2019 16.8

Ideally, we would like to use the latest version because Roslyn keeps making improvements to AdditionalFile handling.

Reply:
I don't remember what the issue was, specifically, I'm afraid. Though I expect we don't need to support VS 2019 anymore, anyhow, since users should be upgrading to .NET 6+ which will require 2022.

Reply:
Resolving the conversation since I did upgrade to CodeAnalysis 4.0.0.